test/Microsoft.IdentityModel.Tokens.Tests/EncryptingCredentialsTests.cs

```diff
@@ -0,0 +1,198 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using Microsoft.IdentityModel.Tests;
+using Xunit;
+
+#pragma warning disable CS3016 // Arrays as attribute arguments is not CLS-compliant
+
+namespace Microsoft.IdentityModel.Tokens.Tests
+{
+    public class EncryptingCredentialsTests
+    {
+        //public EncryptingCredentials(SecurityKey key, string alg, string enc)
+        [Theory, MemberData(nameof(ConstructorATheoryData))]
+        public void ConstructorA(EncryptingCredentialsTheoryData theoryData)
+        {
+            TestUtilities.WriteHeader($"{this}.ConstructorA", theoryData);
+            var context = new CompareContext($"{this}.ConstructorA, {theoryData.TestId}");
+            try
+            {
+                var encryptingCredentials = new EncryptingCredentials(theoryData.Key, theoryData.Alg, theoryData.Enc);
+                theoryData.ExpectedException.ProcessNoException();
+            }
+            catch (Exception exception)
+            {
+                theoryData.ExpectedException.ProcessException(exception);
+            }
+
+            TestUtilities.AssertFailIfErrors(context);
+        }
+
+        //Used in scenarios when a key represents a 'shared' symmetric key
+        //public EncryptingCredentials(SecurityKey key, string enc)
+        [Theory, MemberData(nameof(ConstructorBTheoryData))]
+        public void ConstructorB(EncryptingCredentialsTheoryData theoryData)
+        {
+            TestUtilities.WriteHeader($"{this}.ConstructorB", theoryData);
+            var context = new CompareContext($"{this}.ConstructorB, {theoryData.TestId}");
+            try
+            {
+                var encryptingCredentials = new EncryptingCredentials(theoryData.Key, theoryData.Enc);
+                //Algorithm value should be 'None'
+                IdentityComparer.AreEqual(encryptingCredentials.Alg, SecurityAlgorithms.None, context);
+                theoryData.ExpectedException.ProcessNoException();
```

Comment:
Pass context to ProcessNoException.
theoryData.ExpectedException.ProcessNoException(context);